src/syrupy/location.py

```diff
@@ -114,4 +114,9 @@ def matches_snapshot_name(self, snapshot_name: str) -> bool:
         return self.__parse(self.snapshot_name) == self.__parse(snapshot_name)
 
     def matches_snapshot_location(self, snapshot_location: str) -> bool:
-        return self.filename in snapshot_location
+        loc = Path(snapshot_location)
+        # "test_file" should match_"test_file.ext" or "test_file/whatever.ext", but not
+        # "test_file_suffix.ext"
+        return self.filename == loc.stem or any(
+            self.filename == parent.name for parent in loc.parents
+        )
```

Comment:
I assume this is to handle the SingleFileExtension (image + json). Right now it's not strict enough since the parents will traverse all the way up to the root of the file system.

My proposed logic:

```suggestion
        return self.filename == loc.stem or self.filename == loc.parent.name
```

This does break the ability for an extension to have additional depth in their snapshot directories, but I think that's an okay compromise.

Reply:
Sounds good to me 👍 I've made that update and validated it in the tests.